### PR TITLE
Kissaki

### PR DESCRIPTION
**Pull request by izuku-sds**

### CODE REVIEW

File: config.sample.toml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-kubeconfig = ""
+kubeconfig = "/home/iiteens/.kube/config"
```
Comment (ashpect) [inline config.sample.toml]: don't push. add in gitignore

File: kubernetes/manifests/_setup.yml [hunk 2]
```diff
@@ -35,7 +35,7 @@ metadata:
   name: cluster-admin-role
   namespace: katana
 rules:
-  - apiGroups: ["apps"]
+  - apiGroups: ["apps",""]
```
Comment (ashpect) [inline kubernetes/manifests/_setup.yml:38]: Security vulnerability to give access to all the default core groups. Why was this added ?

File: kubernetes/manifests/kissaki.yml [hunk 3]
```diff
@@ -28,7 +28,8 @@ spec:
       serviceAccountName: cluster-admin
       containers:
         - name: kissaki
-          image: harbor.katana.local/katana/kissaki
+          image: iiteens/kissaki
```
Comment (ashpect) [inline kubernetes/manifests/kissaki.yml]: ??

File: kubernetes/manifests/kissaki.yml [hunk 4]
```diff
@@ -60,4 +61,4 @@ spec:
   ports:
     - protocol: TCP
       port: 80
-      targetPort: 80
+      targetPort: 5000
```
Comment (ashpect) [inline kubernetes/manifests/kissaki.yml]: Kissaki port in katana-services is in 5000, add this to docs to main consistency or add a comment for the same

File: kubernetes/templates/runtime/cc.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      app: {{.AppName}}
+  template:
+    metadata:
+      labels:
+        app: {{.AppName}}
+    spec:
+      serviceAccountName: cluster-admin
+      containers:
+      - name: {{.AppName}}
+        # image: "harbor.katana.local/katana/"{{.Image}}
```
Comment (ashpect) [inline kubernetes/templates/runtime/cc.yml]: revert back

File: lib/deployment/deployment.go [hunk 6]
```diff
@@ -213,3 +214,63 @@ func DeployChallengeToCluster(challengeName, teamName string, firstPatch bool, r
...
 	return nil
 }
+
+
+func DeployChallengeCheckerToCluster(challengeCheckerName, namespace string, replicas int32) error {
+
+	kubeclient, _ := utils.GetKubeClient()
+
+	deploymentsClient := kubeclient.AppsV1().Deployments(namespace)
+	// imageName := "harbor.katana.local/katana/" + challengeCheckerName
```
Comment (ashpect) [inline lib/deployment/deployment.go]: revert back

File: services/challengedeployerservice/controller.go [hunk 7]
```diff
@@ -80,7 +99,136 @@ func Deploy(c *fiber.Ctx) error {
 			}
 			copyChallengeIntoTsuka(challengePath, folderName, challengeType)
 			copyFlagDataIntoKashira(challengePath, folderName)
-			copyChallengeCheckerIntoKissaki(challengePath, folderName)
+			// copyChallengeCheckerIntoKissaki(challengePath, folderName)
```
Comment (ashpect) [inline services/challengedeployerservice/controller.go]: remove

File: services/challengedeployerservice/controller.go [hunk 8]
```diff
@@ -314,3 +462,24 @@ func DeleteChallenge(c *fiber.Ctx) error {
 
 	return c.SendString("Deleted challenge" + challengeName + "in all namespaces.")
 }
+
+// func Deploy(c *fiber.Ctx) error {
```
Comment (ashpect) [inline services/challengedeployerservice/controller.go]: remove

**Comment by ashpect** [comment]
1. Remove testing functions and irrelevant comments
2. Allowing pushing sample challenges for now, we'll create a submodule for it later

**Review by ashpect — COMMENTED** [review]
left comments

File: kubernetes/manifests/_setup.yml (hunk 2 above)
Comment (izuku-sds) [reply kubernetes/manifests/_setup.yml:38]: find its alternative later while testing

File: kubernetes/manifests/kissaki.yml (hunk 3 above)
Comment (izuku-sds) [reply kubernetes/manifests/kissaki.yml:31]: image should be downloaded from harbor only, this was for speedy testing. changed to actual value in new commit.

File: kubernetes/manifests/kissaki.yml (hunk 4 above)
Comment (izuku-sds) [reply kubernetes/manifests/kissaki.yml:64]: comment added

File: services/challengedeployerservice/controller.go (hunk 7 above)
Comment (izuku-sds) [reply services/challengedeployerservice/controller.go:102]: removed this function from functions which I created, later while testing re-evaluate this code in other functions.

File: config.sample.toml (hunk 1 above)
Comment (izuku-sds) [reply config.sample.toml]: reverted changes in config.sample.toml, 
config.toml already in .gitignore

**Review by izuku-sds — COMMENTED** [review]
ashpect's reviews resolved in next commit

File: config.sample.toml (hunk 1 above)
Comment (izuku-sds) [reply config.sample.toml]: > don't push. add in gitignore

reverted changes in config.sample.toml,
config.toml already in .gitignore

File: kubernetes/manifests/_setup.yml (hunk 2 above)
Comment (izuku-sds) [reply kubernetes/manifests/_setup.yml:38]: > Security vulnerability to give access to all the default core groups. Why was this added ?

find its alternative later while testing

File: kubernetes/manifests/kissaki.yml (hunk 3 above)
Comment (izuku-sds) [reply kubernetes/manifests/kissaki.yml]: > image should be downloaded from harbor only, this was for speedy testing. changed to actual value in new commit.

image should be downloaded from harbor only, this was for speedy testing. changed to actual value in new commit.

File: kubernetes/manifests/kissaki.yml (hunk 4 above)
Comment (izuku-sds) [reply kubernetes/manifests/kissaki.yml]: > Kissaki port in katana-services is in 5000, add this to docs to main consistency or add a comment for the same

comment added

File: services/challengedeployerservice/controller.go (hunk 7 above)
Comment (izuku-sds) [reply services/challengedeployerservice/controller.go]: > remove

removed this function from functions which I created, later while testing re-evaluate this code in other functions.
